src/scores/contingency.py

```diff
@@ -0,0 +1,258 @@
+'''
+This foundation class provides an underpinning data structure which can be used for 
+contingency tables of various kinds, also known as a confusion matrix.
+
+It allows the careful setting out of when and where forecasts closely match
+observations.
+
+The simplest kind of table is an accuracy table, which is based on when the 
+forecast and the obs match exactly or not. As real-world data is real-varying, this will
+typically have been calculated as a binary flag which indicates when the forecast
+and observed values are *sufficiently close* based on some tolerance.
+
+The second-simplest kind of table is a binary contingency table which captures true
+positives, true negatives, false positives and false negatives. 
```

Comment:
At the moment, these are currently internal variables only. I have included alias methods on the class where there is more than one score name sharing the same definition. I can introduce internal variable aliases as well, but it seems un-necessary unless you think people are going to want to access raw counts rather instead of using the metrics.